### ISSUE TITLE
Installation problem qtsetup fails

### ISSUE THREAD

**Issue by woodyjon** [therecipe/qt#541]
Hello,

I am trying to install the binding but I have the following errors. Your help would be much appreciated. Here are the steps I did:

- Ubuntu 16.04
- Regular install. I installed QT 5.10.0
- $GOPATH/bin/qtsetup gives me the following errors at step 0062:

I thought it was my gopath not set, but I think it is fine.

```
INFO[0061] generating full qt/speech                    
INFO[0061] generating full qt/quickcontrols2            
INFO[0061] generating full qt/sailfish                  
INFO[0061] generating full qt/webview                   
INFO[0061] generating full qt/remoteobjects             
INFO[0062] running: 'qtsetup install linux' [docker=false] [vagrant=false] 
INFO[0062] installing full qt/core                      
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/core" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install core"

INFO[0062] installing full qt/androidextras             
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/androidextras" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install androidextras"

INFO[0062] installing full qt/gui                       
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/gui" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install gui"

INFO[0062] installing full qt/network                   
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/network" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install network"

INFO[0062] installing full qt/xml                       
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/xml" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install xml"

INFO[0062] installing full qt/dbus                      
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/dbus" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install dbus"

INFO[0062] installing full qt/nfc                       
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/nfc" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install nfc"

INFO[0062] installing full qt/script                    
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/script" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install script"

INFO[0062] installing full qt/sensors                   
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/sensors" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install sensors"

INFO[0062] installing full qt/positioning               
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/positioning" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install positioning"

INFO[0062] installing full qt/widgets                   
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/widgets" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install widgets"

INFO[0062] installing full qt/sql                       
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/sql" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install sql"

INFO[0062] installing full qt/qml                       
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/qml" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install qml"

INFO[0062] installing full qt/websockets                
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/websockets" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install websockets"

INFO[0062] installing full qt/xmlpatterns               
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/xmlpatterns" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install xmlpatterns"

INFO[0062] installing full qt/bluetooth                 
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/bluetooth" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install bluetooth"

INFO[0062] installing full qt/webchannel                
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/webchannel" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install webchannel"

INFO[0062] installing full qt/svg                       
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/svg" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install svg"

INFO[0062] installing full qt/multimedia                
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/multimedia" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install multimedia"

INFO[0062] installing full qt/quick                     
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/quick" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install quick"

INFO[0062] installing full qt/help                      
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/help" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install help"

INFO[0062] installing full qt/location                  
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/location" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install location"

INFO[0062] installing full qt/scripttools               
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/scripttools" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install scripttools"

INFO[0062] installing full qt/uitools                   
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/uitools" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install uitools"

INFO[0062] installing full qt/webengine                 
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/webengine" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install webengine"

INFO[0062] installing full qt/testlib                   
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/testlib" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install testlib"

INFO[0062] installing full qt/serialport                
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/serialport" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install serialport"

INFO[0062] installing full qt/serialbus                 
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/serialbus" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install serialbus"

INFO[0062] installing full qt/printsupport              
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/printsupport" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install printsupport"

INFO[0062] installing full qt/designer                  
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/designer" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install designer"

INFO[0062] installing full qt/scxml                     
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/scxml" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install scxml"

INFO[0062] installing full qt/gamepad                   
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/gamepad" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install gamepad"

INFO[0062] installing full qt/purchasing                
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/purchasing" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install purchasing"

INFO[0062] installing full qt/datavisualization         
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/datavisualization" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install datavisualization"

INFO[0062] installing full qt/charts                    
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/charts" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install charts"

INFO[0062] installing full qt/speech                    
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/speech" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install speech"

INFO[0062] installing full qt/quickcontrols2            
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/quickcontrols2" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install quickcontrols2"

INFO[0062] installing full qt/sailfish                  
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/sailfish" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install sailfish"

INFO[0062] installing full qt/webview                   
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/webview" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install webview"

INFO[0062] installing full qt/remoteobjects             
ERRO[0062] failed to run command                         cmd="go install -p 8 -v github.com/therecipe/qt/remoteobjects" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmdOptionalError name="install remoteobjects"

WARN[0062] failed to install:                           
WARN[0062] core                                         
WARN[0062] androidextras                                
WARN[0062] gui                                          
WARN[0062] network                                      
WARN[0062] xml                                          
WARN[0062] dbus                                         
WARN[0062] nfc                                          
WARN[0062] script                                       
WARN[0062] sensors                                      
WARN[0062] positioning                                  
WARN[0062] widgets                                      
WARN[0062] sql                                          
WARN[0062] qml                                          
WARN[0062] websockets                                   
WARN[0062] xmlpatterns                                  
WARN[0062] bluetooth                                    
WARN[0062] webchannel                                   
WARN[0062] svg                                          
WARN[0062] multimedia                                   
WARN[0062] quick                                        
WARN[0062] help                                         
WARN[0062] location                                     
WARN[0062] scripttools                                  
WARN[0062] uitools                                      
WARN[0062] webengine                                    
WARN[0062] testlib                                      
WARN[0062] serialport                                   
WARN[0062] serialbus                                    
WARN[0062] printsupport                                 
WARN[0062] designer                                     
WARN[0062] scxml                                        
WARN[0062] gamepad                                      
WARN[0062] purchasing                                   
WARN[0062] datavisualization                            
WARN[0062] charts                                       
WARN[0062] speech                                       
WARN[0062] quickcontrols2                               
WARN[0062] sailfish                                     
WARN[0062] webview                                      
WARN[0062] remoteobjects                                
INFO[0062] running: 'qtsetup test linux' [docker=false] [vagrant=false] 
INFO[0062] testing quick/calc                           
ERRO[0062] failed to run command                         cmd="go list std" env= error="exec: \"go\": executable file not found in $PATH" func=RunCmd name="go list std"
```

**Comment by RadhiFadlillah** [comment]
It seems `go` is not detected on your `$PATH`. Could you show the output of :

- `which go`
- `qtsetup check`

**Comment by woodyjon** [comment]
There you go:

```
$ which go
/usr/local/go/bin/go
$ qtsetup check
INFO[0000] running: 'qtsetup check linux' [docker=false] [vagrant=false] 
INFO[0000] GOOS:                        'linux'         
INFO[0000] GOARCH:                      'amd64'         
INFO[0000] GOVERSION:                   'go1.9.2'       
INFO[0000] GOROOT:                      '/usr/local/go' 
INFO[0000] GOPATH:                   *  '/home/jon/Dropbox/go' 
INFO[0000] GOBIN:                    *  '/home/jon/Dropbox/go/bin' 
INFO[0000] QT_HASH:                     'a76e7081468b0d9d554349b66b4971929f036ce7' 
INFO[0000] QT_VERSION:                  '5.10.0'        
INFO[0000] QT_VERSION_MAJOR:            '5.10.0'        
INFO[0000] QT_DIR:                      '/home/jon/Qt'  
INFO[0000] QT_STUB:                     'false'         
INFO[0000] QT_DEBUG:                    'false'         
INFO[0000] QT_QMAKE_DIR:                ''              
INFO[0000] QT_WEBKIT:                   'false'         
INFO[0000] QT_DISTRO:                   'ubuntu'        
INFO[0000] QT_PKG_CONFIG:               'false'   
```

**Comment by hexvalid** [comment]
+1 Same problem

**Comment by gollariel** [comment]
```
export CGO_CXXFLAGS_ALLOW=".*" 
export CGO_LDFLAGS_ALLOW=".*" 
export CGO_CFLAGS_ALLOW=".*" 
$GOPATH/bin/qtsetup
```

This is helped for me

**Comment by woodyjon** [comment]
I understood my problem. It was a problem with the GOPATH.
Basically, as when I launched qtsetup, it was telling me it could not remove files:
```
INFO[0000] running: 'qtsetup prep'                      
PANI[0000] failed to remove /usr/local/bin/qtrcc         error="remove /usr/local/bin/qtrcc: permission denied"
```
So I ran it with sudo. But running it with sudo means it did not take my user environment variables, but the global ones.
So I removed manually the files it could not remove, then I ran qtsetup without sudo, and it worked fine.
